google-cloud-bigtable-emulator/pom.xml

```diff
@@ -147,6 +148,14 @@
       <scope>test</scope>
     </dependency>
 
+  <!-- This is only needed when we're running tests in native image mode-->
+    <dependency>
+      <groupId>com.google.api</groupId>
+      <artifactId>gax-grpc</artifactId>
+      <scope>test</scope>
+    </dependency>
+    <!-- This is only needed when we're running tests in native image mode-->
+
```

Comment:
Should this be put in a separate graalvm specific profile ?

Reply:
Memo:

```
~/cloud-opensource-java $ jar tf ~/Downloads/gax-grpc-2.16.0.jar |grep -i netty                      git[branch:gcb_0.173.0]
com/google/api/gax/grpc/nativeimage/GrpcNettyFeature.class
com/google/api/gax/grpc/nativeimage/substitutions/NettyInternalLoggerFactorySubstitutions$OnlyIfInClassPath.class
com/google/api/gax/grpc/nativeimage/substitutions/NettyInternalLoggerFactorySubstitutions.class
```

Reply:
Thanks for pointing this out @igorbernstein2. Just double checked this and it turns out that these native image configurations (as listed by @suztomo) from gax are needed by the actual application class, `BigtableEmulatorRule`. We need this to be shipped as part of the library. Here is the stacktrace that indicated this:
 
```
  JUnit Vintage:ITBigtableEmulatorRuleTest:testTableAdminClient
    MethodSource [className = 'com.google.cloud.bigtable.emulator.v2.ITBigtableEmulatorRuleTest', methodName = 'testTableAdminClient', methodParameterTypes = '']
    => java.lang.NoClassDefFoundError: Could not initialize class io.grpc.netty.shaded.io.grpc.netty.NettyChannelBuilder
       io.grpc.netty.shaded.io.grpc.netty.NettyChannelProvider.builderForAddress(NettyChannelProvider.java:38)
       io.grpc.netty.shaded.io.grpc.netty.NettyChannelProvider.builderForAddress(NettyChannelProvider.java:24)
       io.grpc.ManagedChannelBuilder.forAddress(ManagedChannelBuilder.java:39)
       com.google.cloud.bigtable.emulator.v2.Emulator.newChannelBuilder(Emulator.java:289)
       com.google.cloud.bigtable.emulator.v2.Emulator.getAdminChannel(Emulator.java:216)
       com.google.cloud.bigtable.emulator.v2.BigtableEmulatorRule.getAdminChannel(BigtableEmulatorRule.java:78)
       com.google.cloud.bigtable.emulator.v2.ITBigtableEmulatorRuleTest.setUp(ITBigtableEmulatorRuleTest.java:48)
       java.lang.reflect.Method.invoke(Method.java:566)
       org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
       org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
       [...]
```

Reply:
I'm very confused: test deps dont get shipped (ie they arent transitive), they are only used when executing tests in current module. So if you want them shipped, you need to mark them as runtime scope

Reply:
Yes, that is a very valid thought! I think this dependency has already been moved out of `test` scope into `compile` scope. Ahead-of-time compilation allows one to compile their application into a lightweight native executable during compile time. During this process, the compiler does a static analysis of all the reachable code in app so this is really when we would need the configurations that are brought in by gax-grpc.